src/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { NextFunction, Request, Response } from "express";
+import fs from 'fs/promises'
+import Joi from "joi";
+
+export const getUserIdFromHeaders = (req: Request): number | null => {
+    const userId = req.headers['x-user-id'];
+
+    if (userId) {
+        return +userId;
+    }
+
+    return null;
+}
+
+export const readFromFile = async <T>(filePath: string): Promise<T> => {
+    return await fs.readFile(filePath).then(data => {
+        return JSON.parse(data.toString()) as T
+    });
+}
+
+export const writeToFile = async <T>(filePath: string, data: T): Promise<void> => {
+    return await fs.writeFile(filePath, JSON.stringify(data))
+}
+
+export const validate = (req: Request, res: Response, next: NextFunction) => {
```

Comment:
I would reccomend:

1. Name the function e.g. `checkUserId` or smth like this instead of `validate`
2. Create `middleware` folder -> `auth.middleware.ts` file in that folder -> move this function into that file